x/rollapp/keeper/msg_server_update_state.go

```diff
@@ -14,39 +12,23 @@ import (
 func (k msgServer) UpdateState(goCtx context.Context, msg *types.MsgUpdateState) (*types.MsgUpdateStateResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
-	if !k.RollappsEnabled(ctx) {
-		return nil, types.ErrRollappsDisabled
-	}
-
 	// load rollapp object for stateful validations
-	rollapp, isFound := k.GetRollapp(ctx, msg.RollappId)
+	_, isFound := k.GetRollapp(ctx, msg.RollappId)
 	if !isFound {
 		return nil, types.ErrUnknownRollappID
 	}
 
-	// check rollapp version
-	if rollapp.Version != msg.Version {
-		return nil, errorsmod.Wrapf(types.ErrVersionMismatch, "rollappId(%s) current version is %d, but got %d", msg.RollappId, rollapp.Version, msg.Version)
-	}
-
 	// call the before-update-state hook
+	// checks if:
+	// 	1. creator is the sequencer
+	// 	2. sequencer rollappId matches the rollappId
+	// 	3. sequencer is bonded
+	// 	4. sequencer is the proposer
```

Comment:
I agree. I only documented the things that take place in the hook for convenience, for not having to look it up. 
I'm also not sure what's the reason to use hooks to check these things. Is it because it's outside of the current module?